src/main/java/org/opensearch/neuralsearch/query/AgenticSearchQueryBuilder.java

```diff
@@ -190,4 +193,28 @@ public String getWriteableName() {
     public String fieldName() {
         return NAME;
     }
+
+    /**
+     * Sanitizes query text to prevent prompt injection attacks
+     */
+    private static String sanitizeQueryText(String queryText) {
+        if (queryText == null) {
+            return null;
+        }
+
+        // Remove potential prompt injection patterns
+        String sanitized = queryText
+            // Remove system/instruction keywords that could manipulate LLM behavior
+            .replaceAll("(?i)\\b(system|instruction|prompt|ignore|forget|override)\\s*:", "")
+            // Remove potential command injection patterns
+            .replaceAll("(?i)\\b(execute|run|eval|script)\\s*[:\\(]", "")
+            .trim();
+
+        // Validate length to prevent extremely long inputs
+        if (sanitized.length() > 1000) {
```

Comment:
nit: should be a static constant, and reference constant in exception message as well